pkg/util/k8s/node.go

```diff
@@ -19,29 +19,40 @@ import (
 	"net"
 
 	v1 "k8s.io/api/core/v1"
+
+	"antrea.io/antrea/pkg/util/ip"
 )
 
-// GetNodeAddr gets the available IP address of a Node. GetNodeAddr will first try to get the NodeInternalIP, then try
+// GetNodeAddrs gets the available IP addresses of a Node. GetNodeAddrs will first try to get the NodeInternalIP, then try
 // to get the NodeExternalIP.
-// Note: Although K8s supports dual-stack, there is only a single Internal address per Node because of issue (
-// kubernetes/kubernetes#91940 ). The Node might have multiple addresses after the issue is fixed, and one per address
-// family. And we should change the return type at that time.
-func GetNodeAddr(node *v1.Node) (net.IP, error) {
-	addresses := make(map[v1.NodeAddressType]string)
+func GetNodeAddrs(node *v1.Node) (*ip.DualStackIPs, error) {
```

Comment:
Maybe you can add a comment to the function that the returned DualStackIPs must include at least one valid IPv4 or IPv6 address if no error is returned, then clients don't have to handle non IP case when there is no error.

Reply:
Updated.